l1-contracts/test/merkle/Merkle.t.sol

```diff
@@ -5,17 +5,41 @@ pragma solidity >=0.8.18;
 import {Test} from "forge-std/Test.sol";
 
 import {NaiveMerkle} from "./Naive.sol";
+import {Merkle} from "../../src/core/libraries/Merkle.sol";
+import {Errors} from "../../src/core/libraries/Errors.sol";
 import {FrontierMerkle} from "./../../src/core/messagebridge/frontier_tree/Frontier.sol";
 
-contract MerkleTest is Test {
+abstract contract MerkleLibWrapper {
+  function verifyMembershipWrapper(
+    bytes32[] memory _path,
+    bytes32 _leaf,
+    uint256 _index,
+    bytes32 _expectedRoot
+  ) external pure {
+    Merkle.verifyMembership(_path, _leaf, _index, _expectedRoot);
+  }
+}
+
+contract MerkleTest is Test, MerkleLibWrapper {
   NaiveMerkle internal merkle;
   FrontierMerkle internal frontier;
 
+  NaiveMerkle internal testNaiveMerkle;
+
   uint256 public constant DEPTH = 10;
 
   function setUp() public {
-    merkle = new NaiveMerkle(DEPTH);
+    // Set up testFrontier
     frontier = new FrontierMerkle(DEPTH);
+    merkle = new NaiveMerkle(DEPTH);
+
+    // Set up testNaive
+    testNaiveMerkle = new NaiveMerkle(DEPTH);
```

Comment:
Having merkle and testNaiveMerkle next too each other here is confusing. I see that merkle is only used in frontier tree test so I would separate the tests here to Frontier.t.sol and Merkle.t.sol and would mention in Merkle.t.sol that it's testing only the Merkle library.